### PR TITLE
python: support hid on (at least) Linux

### PR DESCRIPTION
**Pull request by mlyle**
use like: `sudo python dronin-getconfig -u 20a0:4250`

Tested lightly with python2 and python3 on Linux.  Part of the manufacturing support for Seppuku FC.

Not perfect (especially performance) / likely to be an area of further work but is useful for basic things now.

### CODE REVIEW

File: python/dronin/telemetry.py [hunk 1]
```diff
@@ -479,6 +479,137 @@ def _do_io(self, finish_time):
...
+
+        if vid is None:
+            try:
+                raw = str.split(vidpid, ':')
+
+                if len(raw) != 2:
+                    raise
+
+                vid = int(raw[0], base=16)
+                pid = int(raw[1], base=16)
```
Comment (tracernz) [inline python/dronin/telemetry.py]: pid should be optional IMO. So you can go `20a0:` just like lsusb etc.

File: python/dronin/telemetry.py [hunk 2]
```diff
@@ -479,6 +479,137 @@ def _do_io(self, finish_time):
...
+
+        # XXX do better things here
+        if dev.is_kernel_driver_active(0):
+            dev.detach_kernel_driver(0)
+        if dev.is_kernel_driver_active(1):
+            dev.detach_kernel_driver(1)
+        if dev.is_kernel_driver_active(2):
+            dev.detach_kernel_driver(2)
+
+        #dev.set_configuration()
```
Comment (tracernz) [inline python/dronin/telemetry.py:533]: ??

File: python/dronin/telemetry.py [hunk 3]
```diff
@@ -479,6 +479,137 @@ def _do_io(self, finish_time):
...
+            if f.bEndpointAddress < 0x80:
+                self.ep_out = f
+            else:
+                self.ep_in = f
+
+        if self.ep_out is None:
+            raise RuntimeError("no output endpoint found")
+
+        if self.ep_in is None:
+            raise RuntimeError("no output endpoint found")
```
Comment (tracernz) [inline python/dronin/telemetry.py]: input

**Review by tracernz — COMMENTED** [review]
Great stuff! :grin:

File: python/dronin/telemetry.py (hunk 2 above)
Comment (mlyle) [reply python/dronin/telemetry.py:533]: So, we're supposed to tell the device to take on a configuration.  Our stuff only supports one.

If we do tell it to change configuration, the OS often re-enumerates it, and then seizes it, which is annoying.

**Review by tracernz — APPROVED** [review]

**Pull request merged**